src/main/java/com/pusher/client/AuthenticationFailureException.java

```diff
@@ -0,0 +1,27 @@
+package com.pusher.client;
+
+/**
+ * Used to indicate an authentication failure.
+ *
+ * @see com.pusher.client.UserAuthenticator
+ */
+public class AuthenticationFailureException extends RuntimeException {
+
+    private static final long serialVersionUID = -7208133561904200801L;
```

Comment:
This should be different for the new class. This number seems to have been automatically generated by a tool.

Note: I don't know where do we need serialisation for these exceptions.

Reply:
It doesn't have to be unique among classes. It only has to be changed when we modify the class (add more fields). `Exception`s are serializable, so some tools warn when there's no `serialVersionUID` even if we're not actually serializing.